types/utils.d.ts

```diff
@@ -20,6 +20,27 @@ declare global {
         : { [K in keyof T]?: DeepPartial<T[K]> };
 
     type NoStringIndex<T> = { [K in keyof T as string extends K ? never : K]: T[K] };
+
+    type DeepRequired<T> = T extends Error
```

Comment:
This has also been added in a different branch/PR, the code being exactly the same